pallets/msa/src/lib.rs

```diff
@@ -131,17 +154,20 @@ impl<T: Config> Pallet<T> {
 	}
 
 	pub fn add_key(msa_id: MessageSenderId, key: &T::AccountId) -> DispatchResult {
-		<KeyOwnerOf<T>>::try_mutate(key, |maybe_msa| {
+		<KeyInfoOf<T>>::try_mutate(key, |maybe_msa| {
 			ensure!(maybe_msa.is_none(), Error::<T>::DuplicatedKey);
 
-			*maybe_msa = Some(msa_id);
+			*maybe_msa =
+				Some(KeyInfo { msa_id, expired: T::BlockNumber::zero(), nonce: Zero::zero() });
 
 			Ok(())
 		})
 	}
 
-	pub fn is_msa_owner(who: &T::AccountId, msa_id: MessageSenderId) -> DispatchResult {
-		ensure!(Self::get_owner_of(&who) == Some(msa_id), Error::<T>::NotMsaOwner);
+	pub fn ensure_msa_owner(who: &T::AccountId, msa_id: MessageSenderId) -> DispatchResult {
+		let info = Self::get_key_info(who).ok_or(Error::<T>::NoKeyExist)?;
```

Comment:
did you mean to use `get_owner_of` here?

Reply:
I did not, but now that you point it out I can use it.